File: BlackJack.ViewModels/PlayerViews/GetAllStepsByPlayerIdPlayerView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.PlayerViews
{
    public class GetAllStepsByPlayerIdPlayerView
    {
        public List<PlayerStepGetAllStepsByPlayerIdPlayerViewItem> PlayerSteps { get; set; }

        public GetAllStepsByPlayerIdPlayerView()
        {
            PlayerSteps = new List<PlayerStepGetAllStepsByPlayerIdPlayerViewItem>();
        }
    }

    public class PlayerStepGetAllStepsByPlayerIdPlayerViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public PlayerGetAllStepsByPlayerIdPlayerView Player { get; set; }
        public GameGetAllStepsByPlayerIdPlayerView Game { get; set; }
    }

    public class PlayerGetAllStepsByPlayerIdPlayerView
    {
        public string PlayerId { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }

    public class GameGetAllStepsByPlayerIdPlayerView
    {
        public Guid GameId { get; set; }
        public GameStateTypeEnumView GameState { get; set; }
    }

}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/player.views/get-all-steps-by-playerId.player.view.ts
import { SuiteType } from '../../enums/suite-type';
import { RankType } from '../../enums/rank-type';
import { GameStateType } from '../../enums/game-state-type';

export class GetAllStepsByPlayerIdPlayerView {
  playerSteps?: Array<PlayerStepGetAllStepsByPlayerIdPlayerViewItem>;

  constructor() {
    this.playerSteps = new Array<PlayerStepGetAllStepsByPlayerIdPlayerViewItem>();
  }
}

export class PlayerStepGetAllStepsByPlayerIdPlayerViewItem {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  player?: PlayerGetAllStepsByPlayerIdPlayerView;
  game?: GameGetAllStepsByPlayerIdPlayerView;

  constructor() {
    this.player = new PlayerGetAllStepsByPlayerIdPlayerView();
    this.game = new GameGetAllStepsByPlayerIdPlayerView();
  }
}

export class PlayerGetAllStepsByPlayerIdPlayerView {
  playerId?: string;
  balance?: number;
  bet?: number;
}

export class GameGetAllStepsByPlayerIdPlayerView {
  gameId?: string;
  gameState?: GameStateType;
}
